### DISCUSSION TITLE
Transcriptions differ between Windows CPU and Debian GPU

### DISCUSSION THREAD

**Discussion by webadmin2021** [openai/whisper#308] [Q&A] [answered]
I transcribed an audio file using the following code on Windows CPU and Debian GPU but got different results. The result produced by Windows matches the audio, but one produced by GPU is totally incorrect.

```
logger.info("Loading Model")
model = whisper.load_model("large")
logger.info("Model Loaded")

result = model.transcribe("audio.mp3")

transcript = result["segments"]
write_srt(transcript,  open("subtitle.srt", "w", encoding="utf-8"))
```

Can anyone offer an advice on what could possibly have gone wrong?

**Comment by jongwook** [comment, marked as answer]
Related: #274 

Nondeterministic behavior may arise from various reasons, most likely from the temperature fallback heuristics. You specify `model.transcribe(..., temperature=0.0)` to disable this, but there might still be numerical discrepancies between devices/implementations which can make the results slightly different.
